instance/models/openedx_instance.py

```diff
@@ -220,10 +220,24 @@ def set_active_vm_dns_records(self):
         self.logger.info("Setting DNS records for active app servers...")
```

Comment:
@pkulkark, the code in this method that loops over the active appservers and sets the DNS records and the code that conditionally removes a DNS record should all be wrapped in a `cache.lock` block, with a lock named something like `set_active_vm_dns_records_<unique identifier for the instance>` so that only the calls to the same method for the same instance are serialized. The unique identifier could be the instance `id`. Unless we use a lock, we cannot guarantee that multiple `set_active_vm_dns_records()` calls are not executed at the same time and that can cause issues.

For the same reason, the same lock should also be used in the `clean_up_appserver_dns_records()` method.

Reply:
Ah I see. Added a lock. Thanks!